### PR TITLE
Fix status on shutdown

### LINKED ISSUE #17239: Add environment variable to configure the server shutdown timeout
The timeout ms of terminus is hardcoded to `1000` in the server shutdown. We should add a new `SERVER_SHUTDOWN_TIMEOUT` environment variable to configure this.

### Discussed in https://github.com/directus/directus/discussions/17237

<div type='discussions-op-text'>

<sup>Originally posted by **onordgren** January 19, 2023</sup>
Hello! We're running directus in k8s and would like to be able to control the timeout period in the terminus handler. From the source code it looks like it's hard coded at 1000ms. Is there some other recommended way of doing this, was thinking of maybe doing it in the `server.stop` hook. Or would you consider adding a configuration option to be able to set the terminus timeout?</div>

### LINKED ISSUE #17241: `/health` returns 200 during shutdown
### Describe the Bug

When the API is shutting down, during the shutdown timeout window, the `/health` endpoint will still run and often return 200, even though the API is known to be unhealthy (as it's shutting down). The `/health` endpoint should return a 503 during the shutdown process.

### To Reproduce

Tricky to repro locally, as you have a second exactly (see #17239) to test this: 

- Stop the API process
- During the second shutdown window, hit `/health`

### Errors Shown

n/a

### What version of Directus are you using?

9.22.4

### What version of Node.js are you using?

18.12.1

### What database are you using?

PG 13

### What browser are you using?

Arc

### How are you deploying Directus?

Locally

### PR DESCRIPTION
**Pull request by Nitwel**
It might be handy do add a `await new Promise((resolve) => {setTimeout(resolve, 10000)})` to the `beforeShutdown()` function after we set `SERVER_ONLINE` to false in order to have a larger time window to test this.

Fixes #17241
Fixes ENG-522
Fixes  #17239
Fixes ENG-521

### CODE REVIEW

File: api/src/server.ts [hunk 1]
```diff
@@ -82,7 +84,7 @@ export async function createServer(): Promise<http.Server> {
 	});
 
 	const terminusOptions: TerminusOptions = {
-		timeout: 1000,
+		timeout: env['SERVER_SHUTDOWN_TIMEOUT'],
```
Comment (br41nslug) [inline api/src/server.ts:87]: should we cast this to a number and perhaps do some sanity checking against `Inifinity` or `NaN`?

File: docs/self-hosted/config-options.md [hunk 2]
```diff
@@ -229,6 +229,7 @@ for example. The format for adding LEVELS values is:
 | --------------------------- | -------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
 | `SERVER_KEEP_ALIVE_TIMEOUT` | Timeout in milliseconds for socket to be destroyed | [server.keepAliveTimeout](https://github.com/nodejs/node/blob/master/doc/api/http.md#serverkeepalivetimeout) |
 | `SERVER_HEADERS_TIMEOUT`    | Timeout in milliseconds to parse HTTP headers      | [server.headersTimeout](https://github.com/nodejs/node/blob/master/doc/api/http.md#serverheaderstimeout)     |
+| `SERVER_SHUTDOWN_TIMEOUT`   | Timeout before the server is forcefully shut down  | 1000                                                                                                         |
```
Comment (azrikahar) [inline docs/self-hosted/config-options.md:232]: In the spirit of clarity and making it consistent with the other two timeout env variables above, lets tweak `Timeout` to `Timeout in milliseconds` 👍

**Review by azrikahar — APPROVED** [review]
Aside from the potential env sanity check and tiny docs update, PR's solution itself LGTM 👍

**Review by paescuj — APPROVED** [review]
LGTM!

**Pull request merged**